sos/plugins/ebpf.py

```diff
@@ -0,0 +1,80 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+import os
+import glob
+import json
+
+
+class eBPF(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+
+    plugin_name = 'ebpf'
+    profiles = ('system', 'kernel', 'network')
+
+    def get_bpftool_prog_ids(self, prog_file):
+        out = []
+        try:
+            prog_data = json.load(open(prog_file))
```

Comment:
This will currently not function, as described in the second commit in #1902 - `exec_cmd()` returns json output, but here we're trying to open it the string as a file bject.